samples/core/Multitenancy/exampleschema/Program.cs

```diff
@@ -0,0 +1,182 @@
+
+if (alternateFactory == null)
+{
+    throw new Exception("Alternate factory should not be null.");
+}
+
+foreach (var tenant in tenants)
+{
+    Console.WriteLine($"Setting tenant to: {tenant}");
+    provider.Tenant = tenant;
+    using var ctx = alternateFactory.CreateDbContext();
+    var contacts = ctx.Contacts.OrderBy(c => c.Name).Select(c => c.Name);
+    var contactList = string.Join(", ", contacts);
+    Console.WriteLine($"Loaded contacts: {contactList}");
+}
+
+namespace ExampleSchema
+{
+    /// <summary>
+    /// Simple tenant service.
+    /// </summary>
+    public class TenantProvider
+    {
+        public string Tenant { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Contact.
+    /// </summary>
+    public class Contact
+    {
+        public Guid Id { get; set; } = Guid.NewGuid();
+        public string Name { get; set; } = string.Empty;
+    }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantAContact : Contact { }
+
+    /// <summary>
+    /// Derived class for tenant-specific support.
+    /// </summary>
+    public class TenantBContact : Contact { }
+
+    /// <summary>
+    /// Using IQueryable.
+    /// </summary>
+    public class ContactContext : DbContext
+    {
+        private readonly string _tenant;
+
+        public ContactContext(
+            TenantProvider provider,
+            DbContextOptions<ContactContext> opts)
+            : base(opts) => _tenant = provider.Tenant;
+
+        public DbSet<TenantAContact> TenantAContacts { get; set; } = null!;
+        public DbSet<TenantBContact> TenantBContacts { get; set; } = null!;
+
+        public IQueryable<Contact> Contacts => _tenant == "TenantA" ?
+            TenantAContacts : TenantBContacts;
+
+        // SQLite doesn't handle multiple schemas, but for SQLServer you
+        // can uncomment the code below for TenantaA.Contact and TenantB.Contact
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            //modelBuilder.Entity<TenantAContact>()
+            //    .ToTable(nameof(Contact), "TenantA");
+
+            //modelBuilder.Entity<TenantBContact>()
+            //    .ToTable(nameof(Contact), "TenantB");
+
+            base.OnModelCreating(modelBuilder);
+        }
+    }
+
+    /// <summary>
+    /// Using multiple models with schema/name overrides.
+    /// </summary>
+    public class AlternateContactContext : DbContext
+    {
+        public string Tenant { get; set; } = null!;
+
+        public AlternateContactContext(
+            TenantProvider provider,
+            DbContextOptions<AlternateContactContext> opts)
+            : base(opts) => Tenant = provider.Tenant;
+
+        public DbSet<Contact> Contacts { get; set; } = null!;
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+         =>
+            // SQLite doesn't handle multiple schemas, so we use different
```

Comment:
Yeah, I'd definitely do this sample on SQL Server (LocalDB) so it actually shows what we want to show...